Mathlib/Analysis/Complex/UpperHalfPlane/Metric.lean

```diff
@@ -384,4 +384,47 @@ instance : IsometricSMul SL(2, ℝ) ℍ :=
       exact
         (isometry_real_vadd w).comp (h₀.comp <| (isometry_real_vadd v).comp <| isometry_pos_mul u)⟩
 
+section slices
+
+/--The verticel strip of width A and height B-/
+def upperHalfPlaneSlice (A B : ℝ) :=
+  {z : ℍ | Complex.abs z.1.1 ≤ A ∧ Complex.abs z.1.2 ≥ B}
+
+theorem slice_mem_iff (A B : ℝ) (z : ℍ) :
+    z ∈ upperHalfPlaneSlice A B ↔ Complex.abs z.1.1 ≤ A ∧ Complex.abs z.1.2 ≥ B := Iff.rfl
+
+lemma subset_slice_of_isCompact {K : Set ℍ} (hK : IsCompact K) : ∃  A B : ℝ, 0 < B ∧
+    K ⊆ upperHalfPlaneSlice A B  := by
+  by_cases hne : Set.Nonempty K
+  · have hcts : ContinuousOn (fun t =>  t.im) K := by
+       apply Continuous.continuousOn UpperHalfPlane.continuous_im
+    obtain ⟨b, _, HB⟩ :=  IsCompact.exists_isMinOn hK hne hcts
+    let t := (⟨Complex.I, by simp⟩ : ℍ)
```

Comment:
Maybe it's worth adding `UpperHalfPlane.I`?

Reply:
Good idea, I'll do that.